aspnetcore/signalr/authn-and-authz/6.0sample/SignalRAuthenticationSample/DomainRestrictedRequirement.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.SignalR;
+
+namespace SignalRAuthenticationSample;
+
+public class DomainRestrictedRequirement :
+    AuthorizationHandler<DomainRestrictedRequirement, HubInvocationContext>,
+    IAuthorizationRequirement
+{
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        DomainRestrictedRequirement requirement,
+        HubInvocationContext resource)
+    {
+        var name = context.User.Identity!.Name!;
```

Comment:
@BrennanConroy  the ! operator is ugly here. I could use `#nullable disable` but that seems worse.

Reply:
```csharp
var name = context.User.Identity?.Name;
if (!string.IsNullOrEmpty(name) && ...)
{
    // ...
}
```

Reply:
```
if (context.User.Identity != null && 
     !string.IsNullOrEmpty(context.User.Identity.Name))
        {
```

That looks worse to me but it's what I've pushed. An extra level of `{}`

Reply:
I would have combined what I wrote with the if statement that was already there.
```csharp
var name = context.User.Identity?.Name;
if (!string.IsNullOrEmpty(name) && IsUserAllowedToDoThis(resource.HubMethodName, name) &&
    name.EndsWith("@microsoft.com"))
```

Reply:
```
if (context.User.Identity != null &&
  !string.IsNullOrEmpty(context.User.Identity.Name) && 
  IsUserAllowedToDoThis(resource.HubMethodName,
                       context.User.Identity.Name) &&
  context.User.Identity.Name.EndsWith("@microsoft.com"))
{
        context.Succeed(requirement);
    
}
```
Too much in my opinion.